applications/vanilla/views/discussion/helper_functions.php

```diff
@@ -85,11 +85,17 @@ function writeComment($comment, $sender, $session, $currentOffset) {
         $sender->EventArguments['Permalink'] = $permalink;
 
         // Needed in writeCommentOptions()
-        if ($sender->data('Discussion', null) === null) {
+        $discussion = $sender->data('Discussion', null);
+        
+        if ($discussion === null) {
             $discussionModel = new DiscussionModel();
             $discussion = $discussionModel->getID($comment->DiscussionID);
             $sender->setData('Discussion', $discussion);
         }
+        
+        if (($discussion->InsertUserID ?? false) == $comment->InsertUserID) {
```

Comment:
A few things on this line in particular. Are you sure that the discussion coming in is always an object and not an array? We might need a check here due to the inconsistency data is sometimes passed around as, especially since the the discussion itself is in a data array of any sender that could be passed in.

Also `===` instead of `==` please. Comparing numbers that could be 0 with `==` is a recipe for disaster.

Reply:
I guess that's a rare case `val`.